setup2/src/cli/new.rs

```diff
@@ -56,77 +82,76 @@ pub struct NewOpts {
     )]
     pub curve_type: CurveKind,
 
-    #[options(help = "setup the inner or the outer circuit?")]
-    pub is_inner: bool,
-}
+    #[options(
+        help = "the contribution mode",
+        default = "chunked",
+        parse(try_from_str = "contribution_mode_from_str")
+    )]
+    pub contribution_mode: ContributionMode,
 
-pub fn new(opt: &NewOpts) -> anyhow::Result<()> {
-    if opt.is_inner {
-        let circuit = InnerCircuit::<Testnet2Parameters>::blank();
-        generate_params::<AleoInner, _>(opt, circuit)
-    } else {
-        let mut seed: Seed = [0; SEED_LENGTH];
-        rand::thread_rng().fill_bytes(&mut seed[..]);
-        let rng = &mut ChaChaRng::from_seed(seed);
-        let dpc = Testnet2DPC::load(false)?;
-
-        let noop_circuit = dpc
-            .noop_program
-            .find_circuit_by_index(0)
-            .ok_or(DPCError::MissingNoopCircuit)?;
-        let private_program_input = dpc.noop_program.execute_blank(noop_circuit.circuit_id())?;
-
-        let inner_snark_parameters = <Testnet2Parameters as Parameters>::InnerSNARK::setup(
-            &InnerCircuit::<Testnet2Parameters>::blank(),
-            &mut SRS::CircuitSpecific(rng),
-        )?;
-
-        let inner_snark_vk: <<Testnet2Parameters as Parameters>::InnerSNARK as SNARK>::VerifyingKey =
-            inner_snark_parameters.1.clone().into();
-        let inner_snark_proof = <Testnet2Parameters as Parameters>::InnerSNARK::prove(
-            &inner_snark_parameters.0,
-            &InnerCircuit::<Testnet2Parameters>::blank(),
-            rng,
-        )?;
-
-        let circuit =
-            OuterCircuit::<Testnet2Parameters>::blank(inner_snark_vk, inner_snark_proof, private_program_input);
-        generate_params::<AleoOuter, _>(opt, circuit)
-    }
+    #[options(help = "the chunk size")]
+    pub chunk_size: usize,
+
+    #[options(help = "the size of batches to process", default = "256")]
+    pub batch_size: usize,
+
+    #[options(help = "setup the inner or the outer circuit?", default = "true")]
+    pub is_inner: String,
+
+    #[options(help = "the provided challenge file", default = "challenge")]
+    pub challenge_fname: String,
+    #[options(help = "the new challenge file hash", default = "challenge.verified.hash")]
+    pub challenge_hash_fname: String,
+    #[options(help = "the provided response file which will be verified", default = "response")]
+    pub response_fname: String,
+    #[options(
+        help = "the new challenge file which will be generated in response",
+        default = "new_challenge"
+    )]
+    pub new_challenge_fname: String,
+    #[options(help = "phase 1 file name", default = "phase1")]
+    pub phase1_fname: String,
+    #[options(help = "phase 1 powers")]
+    pub phase1_powers: usize,
+    #[options(help = "number of validators")]
+    pub num_validators: usize,
+    #[options(help = "number of epochs")]
+    pub num_epochs: usize,
```

Comment:
I copied this from the Celo repo; happy to remove it if it ends up being unnecessary.